src/main/java/no/nav/veilarbperson/utils/PersonV2DataMapper.java

```diff
@@ -54,8 +55,25 @@ public static PersonNavnV2 navnMapper(List<HentPerson.Navn> personNavn) {
                 .setForkortetNavn(ofNullable(navn).map(HentPerson.Navn::getForkortetNavn).orElse(null));
     }
 
-    public static Familiemedlem familiemedlemMapper(HentPerson.Familiemedlem familiemedlem, Bostedsadresse personsBostedsadresse) {
+    public static Familiemedlem familiemedlemMapper(HentPerson.Familiemedlem familiemedlem, boolean isEgenAnsatt, Bostedsadresse personsBostedsadresse, AuthService authService) {
         HentPerson.Navn navn = getFirstElement(familiemedlem.getNavn());
+        Fnr medlemFnr = ofNullable(getFirstElement(familiemedlem.getFolkeregisteridentifikator()))
+                .map(HentPerson.Folkeregisteridentifikator::getIdentifikasjonsnummer).map(Fnr::of).orElse(null);
+        String gradering = ofNullable(getFirstElement(familiemedlem.getAdressebeskyttelse())).map(HentPerson.Adressebeskyttelse::getGradering).orElse(null);
+        boolean harVeilederLeseTilgang = authService.harLesetilgang(medlemFnr);
+
+        if (("FORTROLIG".equals(gradering) || "STRENGT_FORTROLIG".equals(gradering) || "STRENGT_FORTROLIG_UTLAND".equals(gradering)) && !harVeilederLeseTilgang) {
+            return new Familiemedlem()
+                .setFodselsnummer(ofNullable(getFirstElement(familiemedlem.getFolkeregisteridentifikator()))
```

Comment:
Kode for å hente fnr er duplisert på linje 72. Kunne hentet fnr før IFen siden den skal brukes begge steder. Eller en hjelpefunksjon som forhindrer duplisering av logikken